domain/src/test/java/org/oppia/android/domain/platformparameter/PlatformParameterSingletonTest.kt

```diff
@@ -0,0 +1,142 @@
+const val stringPlatformParameterValue = "string_platform_parameter_value"
+
+const val integerPlatformParameterName = "integer_platform_parameter_name"
+const val integerPlatformParameterValue = 1
+
+const val booleanPlatformParameterName = "boolean_platform_parameter_name"
+const val booleanPlatformParameterValue = true
+
+@RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+@Config(application = PlatformParameterSingletonTest.TestApplication::class)
+class PlatformParameterSingletonTest {
+
+  @Inject
+  lateinit var platformParameterSingleton: PlatformParameterSingleton
+
+  @Before
+  fun setUp() {
+    setUpTestApplicationComponent()
+  }
+
+  @Test
+  fun testSingleton_initPlatformParameterMap_retrieveStringParameter_verifyItsValue() {
+    assertThat(platformParameterSingleton.platformParameterMap.isEmpty()).isTrue()
+
+    platformParameterSingleton.initPlatformParameterMap(makePlatformParameterMap())
+    assertThat(platformParameterSingleton.platformParameterMap.isEmpty()).isFalse()
+
+    val stringPlatformParameter = platformParameterSingleton.getPlatformParameter<String>(
+      stringPlatformParameterName
+    )
+    assertThat(stringPlatformParameter?.value).isEqualTo(stringPlatformParameterValue)
+  }
+
+  @Test
+  fun testSingleton_initPlatformParameterMap_retrieveIntegerParameter_verifyItsValue() {
+    assertThat(platformParameterSingleton.platformParameterMap.isEmpty()).isTrue()
+
+    platformParameterSingleton.initPlatformParameterMap(makePlatformParameterMap())
+    assertThat(platformParameterSingleton.platformParameterMap.isEmpty()).isFalse()
+
+    val integerPlatformParameter = platformParameterSingleton.getPlatformParameter<Int>(
+      integerPlatformParameterName
+    )
+    assertThat(integerPlatformParameter?.value).isEqualTo(integerPlatformParameterValue)
+  }
+
+  @Test
+  fun testSingleton_initPlatformParameterMap_retrieveBooleanParameter_verifyItsValue() {
+    assertThat(platformParameterSingleton.platformParameterMap.isEmpty()).isTrue()
+
+    platformParameterSingleton.initPlatformParameterMap(makePlatformParameterMap())
+    assertThat(platformParameterSingleton.platformParameterMap.isEmpty()).isFalse()
+
+    val booleanPlatformParameter = platformParameterSingleton.getPlatformParameter<Boolean>(
+      booleanPlatformParameterName
+    )
+    assertThat(booleanPlatformParameter?.value).isEqualTo(booleanPlatformParameterValue)
+  }
+
+  @Test
+  fun testSingleton_initPlatformParameterMap_retrieveIncorrectParameter_verifyItsValue() {
+    assertThat(platformParameterSingleton.platformParameterMap.isEmpty()).isTrue()
+
+    platformParameterSingleton.initPlatformParameterMap(makePlatformParameterMap())
+    assertThat(platformParameterSingleton.platformParameterMap.isEmpty()).isFalse()
+
+    val incorrectPlatformParameter = platformParameterSingleton.getPlatformParameter<String>(
+      "incorrectPlatformParameterName"
+    )
+    assertThat(incorrectPlatformParameter).isEqualTo(null)
```

Comment:
I think there's an `isNull()` check you can use here.

Reply:
Sure will do it